test/e2e/instrumentation/monitoring/custom_metrics-stackdriver.go

```diff
@@ -0,0 +1,200 @@
+			framework.Failf("Unexpected metric %s. Only metric %s should be supported", resource.Name, CustomMetricName)
+		}
+	}
+	value, err := customMetricsClient.NamespacedMetrics("default").GetForObject(schema.GroupKind{Group: "", Kind: "Pod"}, "sd-exporter-1", CustomMetricName)
+	if err != nil {
+		framework.Failf("Failed query: %s", err)
+	}
+	if value.Value.Value() != MetricValue1 {
+		framework.Failf("Unexpected metric value for metric %s: expected %v but received %v", CustomMetricName, MetricValue1, value.Value)
+	}
+	filter, err := labels.NewRequirement("name", selection.Equals, []string{"sd-exporter"})
+	if err != nil {
+		framework.Failf("Couldn't create a label filter")
+	}
+	values, err := customMetricsClient.NamespacedMetrics("default").GetForObjects(schema.GroupKind{Group: "", Kind: "Pod"}, labels.NewSelector().Add(*filter), CustomMetricName)
+	if err != nil {
+		framework.Failf("Failed query: %s", err)
+	}
+	if len(values.Items) != 2 {
+		framework.Failf("Expected results for exactly 2 pods, but %v results received", len(values.Items))
+	}
+	for _, value := range values.Items {
+		if (value.DescribedObject.Name == "sd-exporter-1" && value.Value.Value() != MetricValue1) ||
+			(value.DescribedObject.Name == "sd-exporter-2" && value.Value.Value() != MetricValue2) {
+			framework.Failf("Unexpected metric value for metric %s and pod %s: %v", CustomMetricName, value.DescribedObject.Name, value.Value.Value())
+		}
+	}
+
+	framework.ExpectNoError(err)
+}
+
+func createDescriptors(service *gcm.Service, projectId string) error {
+	_, err := service.Projects.MetricDescriptors.Create(fmt.Sprintf("projects/%s", projectId), &gcm.MetricDescriptor{
+		Name:       CustomMetricName,
+		ValueType:  "INT64",
+		Type:       "custom.googleapis.com/" + CustomMetricName,
+		MetricKind: "GAUGE",
+	}).Do()
+	if err != nil {
+		return err
+	}
+	_, err = service.Projects.MetricDescriptors.Create(fmt.Sprintf("projects/%s", projectId), &gcm.MetricDescriptor{
+		Name:       UnusedMetricName,
+		ValueType:  "INT64",
+		Type:       "custom.googleapis.com/" + UnusedMetricName,
+		MetricKind: "GAUGE",
+	}).Do()
+	return err
+}
+
+func createSDExporterPod(cs clientset.Interface) error {
+	_, err := cs.Core().Pods("default").Create(SDExporterPod("sd-exporter-1", "sd-exporter", CustomMetricName, MetricValue1))
+	if err != nil {
+		return err
+	}
+	_, err = cs.Core().Pods("default").Create(SDExporterPod("sd-exporter-2", "sd-exporter", UnusedMetricName, MetricValue2))
+	return err
+}
+
+func createAdapter() error {
+	stat, err := framework.RunKubectl("create", "-f", "https://raw.githubusercontent.com/GoogleCloudPlatform/k8s-stackdriver/master/custom-metrics-stackdriver-adapter/adapter-beta.yaml")
+	framework.Logf(stat)
+	return err
+}
+
+func cleanupDescriptors(service *gcm.Service, projectId string) {
+	_, _ = service.Projects.MetricDescriptors.Delete(fmt.Sprintf("projects/%s/metricDescriptors/custom.googleapis.com/%s", projectId, CustomMetricName)).Do()
```

Comment:
Can it return an error? If so, it'd make sense to log it for debugging purposes.

Reply:
Added logging.